app/pages/lab/visibility.cjsx

```diff
@@ -27,8 +31,22 @@ module.exports = React.createClass
   setterProperty: 'project'
 
   componentDidMount: ->
+    page = if @props.location?.query?.page? then @props.location.query.page else 1
+    @getWorkflowList page
+
+  onPageChange: (page) ->
+    nextQuery = Object.assign {}, @props.location.query, { page }
+    @context.router.push
+      pathname: @props.location.pathname
+      query: nextQuery
+    @getWorkflowList page
+    workflowList = document.getElementById("workflow-list")
+    workflowList?.scrollIntoView()
```

Comment:
How about we put this in a `then` for the promise of the API request so that it only happens after the successful return of the next or previous page of workflows.